django/db/models/options.py

```diff
@@ -531,9 +534,19 @@ def get_field(self, field_name, many_to_many=None):
 
             return field
         except KeyError:
-            # If the app registry is not ready, reverse fields are
-            # unavailable, therefore we throw a FieldDoesNotExist exception.
-            if not self.apps.ready:
+            pass
+
+        if m2m_in_kwargs:
+            # Previous API does not allow searching reverse fields.
+            raise FieldDoesNotExist('%s has no field named %r' % (self.object_name, field_name))
+
+        # If the app registry is not ready, reverse fields are probably
+        # unavailable, but try anyway.
+        if not self.apps.ready:
+            try:
+                # Don't cache results
```

Comment:
It would probably be worth adding a comment or doc string to the top of this function that explains why this helper method is needed.